joinservice/cmd/main.go

```diff
@@ -56,21 +61,34 @@ func main() {
 
 	handler := file.NewHandler(afero.NewOsFs())
 
-	variant, err := variant.FromString(*attestationVariant)
+	kubeClient, err := kubernetes.New()
+	if err != nil {
+		log.With(zap.Error(err)).Fatalf("Failed to create Kubernetes client")
+	}
+
+	attVariant, err := variant.FromString(*attestationVariant)
 	if err != nil {
 		log.With(zap.Error(err)).Fatalf("Failed to parse attestation variant")
 	}
-	validator, err := watcher.NewValidator(log.Named("validator"), variant, handler)
+
+	certCacheClient := certcache.NewClient(log.Named("certcache"), kubeClient, attVariant)
+	cachedCerts, err := certCacheClient.CreateCertChainCache(ctx)
```

Comment:
we shouldn't introduce a dependency to AMD KDS for non-SNP clusters

Reply:
The Attestation-Variant-Specific logic is hidde behind `CreateCertChainCache`. I thought we agreed to move it out of the `main.go` file.

> Better separate the attestation-variant-specific code in joinservice's main.go. (At least move it to an own func.)

Reply:
sorry, I somehow misread the code in `CreateCertChainCache` (although it's simple :D)